src/lat/determinize-lattice-pruned.h

```diff
@@ -155,11 +155,14 @@ struct DeterminizeLatticePhonePrunedOptions {
   bool word_determinize;
   // minimize: if true, push and minimize after determinization.
   bool minimize;
+  // extra parameters for speed 
+  int max_loop;
+  int max_arcs;
   DeterminizeLatticePhonePrunedOptions(): delta(kDelta),
                                           max_mem(50000000),
                                           phone_determinize(true),
                                           word_determinize(true),
-                                          minimize(false) {}
+                                          minimize(false) , max_loop(-1), max_arcs(-1) {}
```

Comment:
In general I prefer to just set the max-mem in order to limit computation.
It's simpler and has much the same effect. That's why I didn't expose those
variables.  See if you can get the same effect from max_mem.
In any case I'd rather not merge this, as it has no effect in any Kaldi
program-- if you use git for your own repository you can keep the changes
locally and any merging will happen automatically.





On Thu, Jul 21, 2016 at 11:49 PM, rickychanhoyin <notifications@github.com>
wrote:

> In src/lat/determinize-lattice-pruned.h
> <https://github.com/kaldi-asr/kaldi/pull/909#discussion_r71833572>:
>
> >    DeterminizeLatticePhonePrunedOptions(): delta(kDelta),
> >                                            max_mem(50000000),
> >                                            phone_determinize(true),
> >                                            word_determinize(true),
> > -                                          minimize(false) {}
> > +                                          minimize(false) , max_loop(-1), max_arcs(-1) {}
>
> I use it somewhere else in my recognizer which directly set the max_arcs
> and max_loops on the DeterminizeLatticePhonePrunedOptions struct . this
> improve the speed on DeterminizeLatticePhonePrunedWrapper
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/909/files/0ec79f8795d1b315e4ca0fcf11af17dce69512f1#r71833572>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu2oNv2kvGVUwOHZEwgBJaktCzTRpks5qYGgAgaJpZM4JR4CO>
> .
>